server/src/test/java/io/crate/types/GeoShapeTypeTest.java

```diff
@@ -223,5 +213,53 @@ public void test_bytes_estimate() throws Exception {
         assertThat(type.valueBytes(normalize.apply(geoJson.get(6)))).isEqualTo(1320);
         assertThat(type.valueBytes(normalize.apply(geoJson.get(7)))).isEqualTo(880);
     }
+
+    @Test
+    public void test_check_equality() throws Exception {
+        for (Map<String, Object> json: geoJson) {
+            assertThat(type.compare(json, json)).isEqualTo(0);
+        }
+    }
```

Comment:
```suggestion
    @Test
    public void test_check_equality() throws Exception {
        UnaryOperator<Map<String, Object>> normalize = geoJson -> {
            return type.implicitCast(GeoJSONUtils.map2Shape(geoJson));
        };
        for (Map<String, Object> json: geoJson) {
            assertThat(type.compare(json, json)).isEqualTo(0);
            assertThat(type.compare(normalize.apply(json), json)).isEqualTo(0);
            assertThat(type.compare(normalize.apply(json), normalize.apply(json))).isEqualTo(0);
        }
    }
```

Can you check with something like this? Looks like this currently fails but I think it should still pass

Reply:
Looks like `GeoJSONUtils.map2Shape(geoJson)` can trip order of coordinates 
and it breaks `i-th with i-th` element comparison.

Reply:
I will go with another approach

Reply:
I pushed another approach (as discussed with @seut) we will do topological equality for shapes but for geometry collections order must be the same. 

Since normalize here might reshuffle geometry collection (after converting to string and parsing it back), I instead added 2 dedicated tests - one for geometry with same but shuffled shapes and one for topological equality (also added a doc clarification/test for that)